db/etcd.go

```diff
@@ -27,499 +27,336 @@ import (
+	c, err := client.New(etcdCfg)
 	if err != nil {
 		log.Fatal(err)
 	}
 	kapi := client.NewKeysAPI(c)
 
-	resp, err := kapi.Get(context.Background(), "/todd/agents", &client.GetOptions{Recursive: true})
+	return &etcdDB{config: cfg, keysAPI: kapi}
+}
 
-	if resp != nil {
+type etcdDB struct {
+	config  config.Config
+	keysAPI client.KeysAPI
+}
 
-		log.Info("Deleting '/todd/agents' key")
-		resp, err = kapi.Delete(context.Background(), "/todd/agents", &client.DeleteOptions{Recursive: true, Dir: true})
+func (etcddb *etcdDB) Init() error {
 
+	_, err := etcddb.keysAPI.Get(context.Background(), "/todd/agents", &client.GetOptions{Recursive: true})
+	if err == nil {
+		log.Info("Deleting '/todd/agents' key")
+		_, err = etcddb.keysAPI.Delete(context.Background(), "/todd/agents", &client.DeleteOptions{Recursive: true, Dir: true})
 		if err != nil {
-			log.Fatal(err)
-		} else {
-			// print common key info
-			log.Printf("Set is done. Metadata is %q\n", resp)
+			return err
 		}
 	}
 
 	// TODO(mierdin): Consider deleting the entire /todd key here, and recreating all of the various subkeys, just to start from scratch.
 	// Shouldn't need any previous data if you restart the server.
+
+	return nil
 }
 
 // SetAgent will ingest an agent advertisement, and update or insert the agent record
 // in the database as needed.
-func (etcddb etcdDB) SetAgent(adv defs.AgentAdvert) {
-
-	etcd_loc := fmt.Sprintf("http://%s:%s", etcddb.config.DB.IP, etcddb.config.DB.Port)
-	cfg := client.Config{
-		Endpoints: []string{etcd_loc},
-		Transport: client.DefaultTransport,
-		// set timeout per request to fail fast when the target endpoint is unavailable
-		HeaderTimeoutPerRequest: time.Second,
-	}
-
-	c, err := client.New(cfg)
-	if err != nil {
-		log.Fatal(err)
-	}
-	kapi := client.NewKeysAPI(c)
-
+func (etcddb *etcdDB) SetAgent(adv defs.AgentAdvert) error {
 	log.Infof("Setting '/todd/agents/%s' key", adv.Uuid)
 
-	adv_json, err := json.Marshal(adv)
+	advJSON, err := json.Marshal(adv)
 	if err != nil {
 		log.Error("Problem converting Agent Advertisement to JSON")
-		os.Exit(1)
+		return err
 	}
 
 	// TODO(mierdin): TTL needs to be user-configurable
-	resp, err := kapi.Set(
+	resp, err := etcddb.keysAPI.Set(
 		context.Background(),                      // context
 		fmt.Sprintf("/todd/agents/%s", adv.Uuid),  // key
-		string(adv_json),                          // value
+		string(advJSON),                           // value
 		&client.SetOptions{TTL: time.Second * 30}, //optional args
 	)
 	if err != nil {
 		log.Error("Problem setting agent in etcd")
-		os.Exit(1)
+		return err
 	}
 
 	log.Infof("Agent set in etcd. This advertisement is good until %s", resp.Node.Expiration)
 
+	return nil
 }
 
 // GetAgent will retrieve a specific agent from the database by UUID
-func (etcddb etcdDB) GetAgent(uuid string) defs.AgentAdvert {
+func (etcddb *etcdDB) GetAgent(uuid string) (*defs.AgentAdvert, error) {
+	log.Printf("Getting /todd/agents/%s' key value", uuid)
```

Comment:
Maybe put ```keyStr``` at the top and reference it in this logging statement instead of repeating the string literal? I know this is being done multiple times in each etcddb function so it might keep things more consistent